cdap-api-common/src/main/java/co/cask/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -108,30 +108,78 @@ private Schema read(JsonReader reader, Map<String, Schema> knownRecords) throws
         return readUnion(reader, knownRecords);
       case BEGIN_OBJECT: {
         reader.beginObject();
-        String name = reader.nextName();
-        if (!"type".equals(name)) {
-          throw new IOException("Property \"type\" missing.");
+        // Type of the schema
+        Schema.Type schemaType = null;
+        // Name of the element
+        String elementName = null;
+        // Store enum values for ENUM type
+        List<String> enumValues = new ArrayList<>();
+        // Store schema for key and value for MAP type
+        Schema keys = null;
+        Schema values = null;
+        // List of fields for RECORD type
+        List<Schema.Field> fields = null;
+        // List of items for ARRAY type
+        Schema items = null;
+        // Loop through current object and populate the fields as required
+        // For ENUM type List of enumValues will be populated
+        // For ARRAY type items will be populated
+        // For MAP type keys and values will be populated
+        // For RECORD type fields will be popuated
+        while (reader.hasNext()) {
+          String name = reader.nextName();
+          switch (name) {
+            case "type":
+              schemaType = Schema.Type.valueOf(reader.nextString().toUpperCase());
+              break;
+            case "name":
+              elementName = reader.nextString();
+              if (schemaType == Schema.Type.RECORD) {
+                knownRecords.put(elementName, null);
+              }
+              break;
+            case "symbols":
+              enumValues = readEnum(reader);
+              break;
+            case "items":
+              items = read(reader, knownRecords);
+              break;
+            case "keys":
+              keys = read(reader, knownRecords);
+              break;
+            case "values":
+              values = read(reader, knownRecords);
+              break;
+            case "fields":
+              fields = getFields(name, reader, knownRecords);
+              knownRecords.put(elementName, Schema.recordOf(elementName, fields));
+              break;
+            default:
+              reader.skipValue();
+              break;
```

Comment:
this is not necessary right? Ok if you want to keep it for readability, but let's add it in the other switch statements in this class too, for consistency.